libs/feature/search/src/lib/elasticsearch/mapper/atomic-operations.ts

```diff
@@ -0,0 +1,47 @@
+import { MetadataLink } from '@geonetwork-ui/util/shared'
+
+export type SourceWithUnknownProps = { [key: string]: unknown }
+
+export const selectField = <T>(
+  source: SourceWithUnknownProps,
+  fieldName: string
+): T | null =>
+  source !== null && fieldName in source ? (source[fieldName] as T) : null
+
+export const selectFallbackFields = <T>(
+  source: SourceWithUnknownProps,
+  ...fieldNames: string[]
+): T | null =>
+  fieldNames.reduce<T>(
+    (prev, curr) => (prev === null ? selectField(source, curr) : prev),
+    null
+  )
+
+export const selectFallback = (field, fallback) =>
+  field === null ? fallback : field
```

Comment:
And if it's a string `undefined` or `''` ? Default value is not applied.

Reply:
in these operations `null` is used when a key is missing; basically the idea is that you can chain operations like `getAsDate(selectFallback(selectField()))` and not risk a runtime error; `null` will just propagate through